### PR TITLE
Feat/scout 11 orders module

### PR DESCRIPTION
**Pull request by dejwi**
[SCOUT-11](https://playmakerpro.atlassian.net/browse/SCOUT-11?atlOrigin=eyJpIjoiOTJjNzQzN2ZlOTMzNDFhMWFkY2ViYmNkMWYwZTQ3NWUiLCJwIjoiaiJ9)

### Task Description

Migrate orders module
- pages
- modules
- both^ without edit
- OrderStatusChip component

### CODE REVIEW

**Review by jakubfiglak — CHANGES_REQUESTED** [review]
As usual, great work 🚀 

There're some adjustments to be made though, please take a look at my in-code comments

File: src/modules/orders/table/row.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      <StyledTableCell>{formatDate(createdAt)}</StyledTableCell>
+      <StyledTableCell padding="checkbox" align="center">
+        {!!description && (
+          <Tooltip title={description}>
+            <NoteIcon />
+          </Tooltip>
+        )}
+      </StyledTableCell>
+      <StyledTableCell align="center">
+        <Badge badgeContent={_count.reports} color="secondary">
```
[the remark this answers is not in the document]
Comment (dejwi) [reply src/modules/orders/table/row.tsx:120]: just a sidenote it worked with count higher than zero and I didnt notice || '0' passed as **string** in v1 🙆‍♂️🙆‍♂️🙆‍♂️🙆‍♂️

[the remark this answers is not in the document]
Comment (jakubfiglak) [reply src/modules/orders/table/row.tsx:120]: Yeah, it looks like this component doesn't display a badge if `badgeContent === 0`. That makes sense if you use it for example to show some notifications count. In our case, we want to display 0, so the string was a hack to do so:)

You can always check the docs - https://mui.com/material-ui/react-badge/

File: src/modules/orders/table/row.tsx [hunk 2]
```diff
@@ -103,9 +96,11 @@ export const OrdersTableRow = ({
           />
         </TableMenu>
       </StyledTableCell>
-      <StyledTableCell>{`${player?.firstName} ${player?.lastName}`}</StyledTableCell>
+      <CellWithLink href={getSinglePlayerRoute(player?.slug || '')} label={`${player?.firstName} ${player?.lastName}`} />
       <StyledTableCell>{player?.primaryPosition.name}</StyledTableCell>
-      <StyledTableCell>{player?.teams[0].team.name}</StyledTableCell>
+      <CellWithLink href={`/teams/${player?.teams[0].team.slug}`} label={player?.teams[0].team.name || ''} />
```
Comment (jakubfiglak) [inline src/modules/orders/table/row.tsx:101]: Please use `getSingleTeamRoute` function here

File: src/modules/orders/table/row.tsx [hunk 3]
```diff
@@ -103,9 +96,11 @@ export const OrdersTableRow = ({
...
+      <CellWithLink href={`/matches/${match?.id}`} label={match ? `${match?.homeTeam.name} vs ${match.awayTeam.name}` : ''} />
```
Comment (jakubfiglak) [inline src/modules/orders/table/row.tsx:102]: Please use `getSingleMatchRoute` function here & `getMatchDisplayName` for the `label`.

I'd like to create these kind of functions for all of our modules to reduce the amount of repeated code (like building labels - we do this stuff all around the app) so please existing functions or write new ones if they've not been written before.

**Review by jakubfiglak — COMMENTED** [review]
We're getting close, just two minor comments:)

File: src/modules/orders/table/row.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            onClick={() => {
+              handleMenuAction(onDeleteClick)
+            }}
+            disabled={!isDeleteOptionEnabled}
+          />
+        </TableMenu>
+      </StyledTableCell>
+      <CellWithLink href={getSinglePlayerRoute(player?.slug || '')} label={player ? `${player?.firstName} ${player?.lastName}` : ''} />
+      <StyledTableCell>{player?.primaryPosition.name}</StyledTableCell>
+      <CellWithLink href={getSingleTeamRoute(player?.teams[0].team.slug || '')} label={player?.teams[0].team.name || ''} />
```
Comment (jakubfiglak) [inline src/modules/orders/table/row.tsx:103]:
```suggestion
        {player?.teams[0] ? (
          <CellWithLink
            href={getSingleTeamRoute(player.teams[0].team.slug}
            label={player.teams[0].team.name}
          />
        ) : (
          <StyledTableCell>-</StyledTableCell>
        )}
```

Please do not render `CellWithLink` component when there's no team data

File: src/modules/orders/table/row.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      <CellWithLink href={getSingleMatchRoute(match?.id || 0)} label={match ? getMatchDisplayName({ awayTeamName: match.awayTeam.name, homeTeamName: match.homeTeam.name, competitionName: match.competition.name }) : ''} />
```
Comment (jakubfiglak) [inline src/modules/orders/table/row.tsx:104]:
```suggestion
        {match ? (
          <CellWithLink
            href={getSingleMatchRoute(match.id)}
            label={getMatchDisplayName({
              homeTeamName: match.homeTeam.name,
              awayTeamName: match.awayTeam.name,
            })}
          />
        ) : (
          <StyledTableCell>-</StyledTableCell>
        )}
```

Same here. And please omit `competitionName` in table cells, it's too much text

File: src/utils/withSessionSsrRole.ts [hunk 6]
```diff
@@ -60,6 +61,7 @@ export function withSessionSsrRole<T>(
       const resd = await getData(
         req.session.token as string,
         params as ParsedUrlQuery,
+        user,
```
Comment (jakubfiglak) [inline src/utils/withSessionSsrRole.ts:64]: I don't remember if I asked this before - why do we need user here?

**Review by jakubfiglak — CHANGES_REQUESTED** [review]
Sorry, some more adjustments :D

File: src/utils/withSessionSsrRole.ts (hunk 6 above)
Comment (dejwi) [reply src/utils/withSessionSsrRole.ts]: It's used in index page at handleSetFilters in order to hack together showing only mine orders (userId query)

Comment (jakubfiglak) [reply src/utils/withSessionSsrRole.ts]: I don't think there's a reason why we should do this server side. We have a `useUser` and we can get current user id using that on the client side. I'm gonna approve this PR anyway but let's keep in mind we should revisit this topic in the future.

**Review by jakubfiglak — APPROVED** [review]

File: src/utils/withSessionSsrRole.ts (hunk 6 above)
Comment (dejwi) [reply src/utils/withSessionSsrRole.ts]: ohh I didn't know about this hook 🙆‍♂️

**Pull request merged**